lib/theme_settings_object_validator.rb

```diff
@@ -94,15 +87,30 @@ def validate
 
   private
 
-  def validate_properties
-    @properties.each do |property_name, property_attributes|
-      next if property_attributes[:type] == "objects"
-      next if property_attributes[:required] && !is_property_present?(property_name)
-      next if !has_valid_property_value_type?(property_attributes, property_name)
-      next if !has_valid_property_value?(property_attributes, property_name)
+  def validate_child_objects(objects, property_name:, schema:)
+    return if objects.blank?
+
+    objects.each_with_index do |object, index|
+      self
+        .class
+        .new(
+          schema:,
+          object:,
+          valid_category_ids:,
+          json_pointer_prefix: "#{@json_pointer_prefix}#{property_name}/#{index}/",
+          errors: @errors,
+        )
+        .validate
     end
   end
 
+  def validate_property(property_name, property_attributes)
+    return if property_attributes[:type] == "objects"
```

Comment:
I think we can delete this line since this method is only called when the type isn't `objects`.

Reply:
O good catch 👍